cmd/juju/commands/upgrademodel.go

```diff
@@ -991,27 +999,26 @@ func (context *upgradeContext) maybeChoosePackagedAgent() (err error) {
 		if found {
 			logger.Debugf("found a more recent stable version %s", newestNextStable)
 			context.chosen = newestNextStable
-		} else {
-			newestCurrent, found := context.packagedAgents.NewestCompatible(context.agent)
-			if found {
-				logger.Debugf("found more recent current version %s", newestCurrent)
-				context.chosen = newestCurrent
-			} else {
-				if context.agent.Major != context.client.Major {
-					return errors.New("no compatible agent versions available")
-				} else {
-					return errors.New("no more recent supported versions available")
-				}
-			}
+			return nil
 		}
-	} else {
-		// If not completely specified already, pick a single tools version.
-		filter := coretools.Filter{Number: context.chosen}
-		if context.packagedAgents, err = context.packagedAgents.Match(filter); err != nil {
-			return errors.Wrap(err, errors.New("no matching agent versions available"))
+		newestCurrent, found := context.packagedAgents.NewestCompatible(context.agent)
+		if found && newestCurrent.Compare(context.agent) > 0 {
```

Comment:
do we need `newestCurrent.Compare(context.agent) > 0`?
Doesn't `NewestCompatible` give us a newestCurrent that is always > context.agent?

Reply:
It could return a version equal to `context.agent`.